lib/_http_outgoing.js

```diff
@@ -382,7 +382,7 @@ OutgoingMessage.prototype.setHeader = function setHeader(name, value) {
 
 
 OutgoingMessage.prototype.getHeader = function getHeader(name) {
-  if (arguments.length < 1) {
+  if (typeof name !== 'string') {
```

Comment:
hmm... I *think* this technically makes this a semver-major because of the change in *why* the error is thrown, although there would be no way of triggering that difference so it likely wouldn't matter. Therefore I'm +1 with this part as semver-patch